### PR TITLE
[toc2] Support custom colors for navigation text and window

### PR DESCRIPTION
**Pull request by hiiwave**
It's a PR for toc2 (Table of Contents). 
The colors for the navigation text and window were hard-coded. 
I added some options to make those colors customizable, 
which is especially useful for users using jupyter-notebook with a dark theme.

### CODE REVIEW

**Comment by hiiwave** [comment]
Here is a demo using *onedork* theme from [jupyter-themes](https://github.com/dunovank/jupyter-themes):

### Before
before

### After
demo

File: src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ Parameters:
 - name: toc2.moveMenuLeft
   description: Move notebook's title and menu on the left instead of being centered -- This provides a better look when the toc/sidebar is present
   input_type: checkbox
-  default: true    
+  default: true
```
Comment (jcb91) [inline src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.yaml]: can we lose the trailing whitespace here please?
In fact, anywhere there is any, I guess :wink:

File: src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.yaml [hunk 2]
```diff
@@ -52,4 +52,19 @@ Parameters:
   input_type: color
   description: Color of sections with running cells
   default: "#FF0000"  
-
+- name: toc2.colors.wrapper_background
+  input_type: color
+  description: Color of wrapper window background
+  default: "#FFF"
```
Comment (jcb91) [inline src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.yaml]: I have a feeling that defaults specified as three-character codes don't load well in the configurator - could we use full six-char codes here?

File: src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.yaml [hunk 3]
```diff
@@ -52,4 +52,19 @@ Parameters:
...
+  description: Color of sidebar border
+  default: "#EEE"
+- name: toc2.colors.navigate_text
+  input_type: color
+  description: Color of navigate text
+  default: "#333"
+- name: toc2.colors.navigate_num
+  input_type: color
+  description: Color of navigate number
+  default: "#000"
```
Comment (jcb91) [inline src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.yaml]: let's have a blank final line :)

**Comment by hiiwave** [comment]
Thanks for the suggestion. Please take a look for the new commit 😄

**Comment by jcb91** [comment]
:+1: looks good to me!

**Comment by jfbercher** [comment]
Nice idea and well done! Thanks. :+1: 
Perhaps can you also edit the README.md to  include a brief description of the new possibilities; something like 
```- Customization of background, fonts, border and highlighting colors in the toc window and navigation menus ```, perhaps also include the gif picture with the modified theme (as above)
and add your contribution at the bottom of the file...

**Comment by hiiwave** [comment]
Sure. Please take a look.

**Comment by hiiwave** [comment]
Also please feel free to edit the readme to make it proper :smile

File: src/jupyter_contrib_nbextensions/nbextensions/toc2/README.md [hunk 4]
```diff
@@ -30,9 +30,13 @@ The initial configuration can be given using the IPython-contrib nbextensions fa
 - The numbering of headers (true by default)
 - Moving header title and menus on the left (default: true)
 - Customization of highlighting the title of currently selected/running sections.  
+- Customization of background, fonts, border and highlighting colors in the toc window and navigation menus. Hence it's possible to be used under notebook with a dark theme. (see thrid demo)
```
Comment (jfbercher) [inline src/jupyter_contrib_nbextensions/nbextensions/toc2/README.md]: Remove the line just before and add a dot at the end.

Comment (jfbercher) [inline src/jupyter_contrib_nbextensions/nbextensions/toc2/README.md]: <strike> thrid</strike> --> *third*

Comment (hiiwave) [inline src/jupyter_contrib_nbextensions/nbextensions/toc2/README.md]: ok.

**Comment by jcb91** [comment]
One last nit-pick request, could you rebate to remove the commits 2ebe344457eaec82e6c34fbdb0eaaf25a49e5d28 and b8ca0c5, just to keep the history slightly smaller? Not a big deal, but we're getting a touch bloated, particularly with all the animated-gif demos 😆

**Comment by hiiwave** [comment]
Done 😄

**Comment by jcb91** [comment]
Thanks! Ok, looks good to merge to me, @jfbercher are we good to go?

**Comment by jfbercher** [comment]
Yes, that's fine.

**Comment by jfbercher** [comment]
Thanks, @hiiwave

**Comment by hiiwave** [comment]
Thank you both too. It's really joyful that I just did my first contribution on an open source project 😄 
Especially thanks @jcb91, I learned a lot from conversation on project `jupyter_highlight_selected_word` with him!

**Comment by marius311** [comment]
Thanks @hiiwave  this is a really awesome thing to have! Unfortunately it doesn't seem to be working for me, I don't see any change to the colors of the sidebar TOC, it looks exactly like your "before" picture rather than your "after". I tried both with this exact commit, as well as current master, so I guess it must be something related to a different package? I have Jupyter notebook 5.0.0 and jupyterthemes 0.16.4.

If its of any help, I noticed that when I delete the <style> tag highlighted below, then it works, although I lose the nice highlighting of the current running cells, etc...

image

**Comment by jcb91** [comment]
@marius311 apologies for the delayed reply, if this is still not working for you, could you open a new issue please?

**Comment by hiiwave** [comment]
@marius311 Sorry that I missed the conservation as I was busy on my thesis then and forgot it thereafter. I have no clue why it did not work out at your side, but I just confirmed that the color still displays properly under jupyter notebook 6.0.0 and jupyterthemes 0.20.0. Perhaps you can create a new clean virtual environment with these versions and check if this works at your side?

**Comment by marius311** [comment]
Thanks for the followup, although I'm mostly using JupterLab now so this is no longer an issue for me.

**Pull request merged**